src/components/assetlib_projects/asset_library_projects.gd

```diff
@@ -0,0 +1,353 @@
+	
+	var body = byte_body.get_string_from_ascii()
+	var json = JSON.new()
+	if json.parse(body) != OK or not json.data is Dictionary:
+		_message(true, tr("Received unexpected data."))
+		_current_assets = {}
+		return
+	_current_assets = json.data
+	_clear_message()
+
+
+func _display_assets(current_assets: Dictionary):
+	_current_assets = current_assets
+	_clear_asset_display()
+	if current_assets == {} or current_assets.total_items == 0:
+		return
+	
+	_display_navigation()
+	
+	var assets = current_assets.result
+	for asset_data in assets:
+		var asset = _ASSET_LISTING.instantiate().init_asset_listing_interactible(
+				int(asset_data.asset_id),
+				asset_data.title,
+				asset_data.category,
+				asset_data.author,
+				asset_data.cost,
+				projects
+		)
+		
+		_asset_list.add_child(asset)
+
+
+func _clear_navigation_buttons():
+	for button in _page_buttons.get_children():
+		button.hide()
+
+
+## Sets up the navigation buttons, taking into account number of pages
+## and current page number.
+func _display_navigation():
+	_clear_navigation_buttons()
+	if not _current_assets or _current_assets.pages == 0:
+		_navigation_buttons.hide()
+		return
+	_navigation_buttons.show()
+	var buttons = _page_buttons.get_children()
+	@warning_ignore("integer_division")
+	var center_offset = _MAX_PAGE_BUTTONS / 2
+	var start_page = max(0, _current_page - center_offset)
+	for i in _MAX_PAGE_BUTTONS:
+		var page = start_page + i
+		if page > _current_assets.pages - 1:
+			break
+		buttons[i].show()
+		buttons[i].text = str(start_page + i + 1)
+		buttons[i].disabled = start_page + i == _current_page
+	
+	if _current_page == 0:
+		_first_button.disabled = true
+		_previous_button.disabled = true
+		_next_button.disabled = false # Assumes more than one page
+		_last_button.disabled = false
+	elif _current_page == _current_assets.pages - 1:
+		_first_button.disabled = false
+		_previous_button.disabled = false
+		_next_button.disabled = true
+		_last_button.disabled = true
+	else:
+		_first_button.disabled = false
+		_previous_button.disabled = false
+		_next_button.disabled = false
+		_last_button.disabled = false
+
+
+## Clears all assets being displayed.
+func _clear_asset_display():
+	_navigation_buttons.hide()
+	for child in _asset_list.get_children():
+		child.queue_free()
+
+
+## Fetches all versions (strings in the MAJOR.MINOR.PATCH format) listed
+## on TuxFamily.  Returns [code][][/code] on failure.
+func _fetch_versions() -> Array[String]:
+	var request = HTTPRequest.new()
+	request.timeout = 8
+	add_child(request)
+	request.request(_TUXFAMILY_VERSION_LISTING)
+	_message(false, tr("Fetching version info…"))
+	var resp = await request.request_completed
+	request.queue_free()
+	if resp[0] != HTTPRequest.RESULT_SUCCESS or resp[1] != 200:
+		_message(true, tr("Version info fetching failed."))
+		return []
+	
+	var body = resp[3].get_string_from_ascii()
+	var regex = RegEx.new()
+	regex.compile("(?<=td class=\\\"n\\\"><a href=\\\")\\d\\.\\d(\\.\\d)?")
```

Comment:
I know we have a class to handle XML under extensions/xml.gd and that godot also offers some built-in functionality for that, but I thought here, a regex is just the simplest solution.